src/piracy.jl

```diff
@@ -77,71 +78,118 @@ is_foreign(@nospecialize(x), pkg::Base.PkgId) = is_foreign(typeof(x), pkg)
 # a crazy API).  The symbol name may also come from `gensym`.  Since the aim of
 # `Aqua.test_piracy` is to detect only "obvious" piracy, let us play on the
 # safe side.
-is_foreign(x::Symbol, pkg::Base.PkgId) = false
+is_foreign(x::Symbol, pkg::Base.PkgId; treat_as_own_type::Vector{<:Type} = Type[]) = false
```

Comment:
Do we ever *not* pass these `treat_as_own`? If not, then why even provide a default value?

Also, just wondering: why use `Vector` and not `Set`? For shorter (or empty) lists it may not matter, but perhaps if they are bigger? For that matter, we could simply not restrict the type here at all?)
```suggestion
is_foreign(x::Symbol, pkg::Base.PkgId, treat_as_own_type::Set)
```